### PR TITLE
Refactoring XMLMessageSerializer

### PR DESCRIPTION
**Pull request by johnsimons**
Have splitted:
- Caching
- Serialization
- Deserialization

### CODE REVIEW

**Comment by johnsimons** [comment]
@Particular/nservicebus thoughts so far?
Any other obvious changes we should do ?

**Comment by johnsimons** [comment]
@Particular/nservicebus let me know if there is any other improvements we can do, otherwise I will merge this on as it is for now.

**Comment by danielmarbach** [comment]
How about moving some of the processing stuff into more descriptive methods?

On Mon, Feb 9, 2015 at 1:23 AM, John Simons <notifications@github.com>
wrote:

> @Particular/nservicebus
> <https://github.com/orgs/Particular/teams/nservicebus> let me know if
> there is any other improvements we can do, otherwise I will merge this on
> as it is for now.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/2647#issuecomment-73442316>
> .
>



-- 

Daniel Marbach

Solution Architect, Particular Software

Email:  daniel.marbach@particular.net

Web:    www.particular.net
<http://t.signauxneuf.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJN7t5XX4RzDjlVS9QB88pTc_dW3Ljg7x56dWyBdgBXs402?t=http%3A%2F%2Fwww.particular.net%2F&si=4896787407044608&pi=33cc6202-1f31-4040-8d6c-e70641f44bba>

**Comment by johnsimons** [comment]
@danielmarbach it can be done, what methods do you think we should rename ?

File: src/NServiceBus.Core/Serializers/XML/Deserializer.cs [hunk 1]
```diff
@@ -0,0 +1,638 @@
...
+                });
+
+            doc.Load(reader);
+
+            if (doc.DocumentElement == null)
+            {
+                return result.ToArray();
+            }
+
+            foreach (XmlAttribute attr in doc.DocumentElement.Attributes)
```
Comment (danielmarbach) [inline src/NServiceBus.Core/Serializers/XML/Deserializer.cs]: Move out to method with descriptive name 59-86

File: src/NServiceBus.Core/Serializers/XML/Deserializer.cs [hunk 2]
```diff
@@ -0,0 +1,638 @@
...
+                        }
+                        else
+                        {
+                            prefixesToNamespaces[prefix] = attr.Value;
+                        }
+                    }
+                }
+            }
+
+            if (doc.DocumentElement.Name.ToLower() != "messages")
```
Comment (danielmarbach) [inline src/NServiceBus.Core/Serializers/XML/Deserializer.cs]: Maybe leave high level if/else here but introduce descriptive method for condition and processing bodies of this if/else branch to line 138

**Comment by danielmarbach** [comment]
@johnsimons made some comments inline to illustrate what I mean

**Pull request merged**